debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9JavaVMHelper.java

```diff
@@ -34,8 +35,34 @@
 import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
 import com.ibm.j9ddr.vm29.pointer.generated.J9MethodPointer;
 import com.ibm.j9ddr.vm29.pointer.generated.J9VMSystemPropertyPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9BuildFlags;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import static com.ibm.j9ddr.vm29.structure.J9Consts.*;
 
 public class J9JavaVMHelper {
+	public static final boolean compressObjectReferences;
+
+	static {
+		try {
+			if (J9ObjectHelper.mixedMode) {
+				J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+				compressObjectReferences = vm.extendedRuntimeFlags2().anyBitsIn(J9_EXTENDED_RUNTIME2_COMPRESS_OBJECT_REFERENCES);
+			} else {
+				compressObjectReferences = J9BuildFlags.gc_compressedPointers;
+			}
+		} catch(CorruptDataException e) {
+			throw new RuntimeException(e);
+		}
+	}
```

Comment:
I think this would be better located in `J9ObjectHelper` - I thought there was an initialization cycle for a minute.

Reply:
Moved.